subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/DefaultTaskCacheabilityResolver.java

```diff
@@ -60,9 +60,9 @@ public Optional<CachingDisabledReason> shouldDisableCaching(
                 }
             }
 
-            Optional<Optional<CachingDisabledReason>> maybeCachingDisabledReason = task.getReasonNotToTrackState()
-                .map(s -> Optional.of(new CachingDisabledReason(CachingDisabledReasonCategory.NOT_CACHEABLE, "Task is untracked because: " + s)));
-            return maybeCachingDisabledReason.orElseGet(() -> Optional.of(CACHING_NOT_ENABLED));
+            return task.getReasonNotToTrackState()
+                .map(s -> Optional.of(new CachingDisabledReason(CachingDisabledReasonCategory.NOT_CACHEABLE, "Task is untracked because: " + s)))
+                .orElseGet(() -> Optional.of(CACHING_NOT_ENABLED));
```

Comment:
🧐 This is nitpicking, but you could also let `Optional.of()` wrap around this chain of calls.